python/cudf/dataframe/dataframe.py

```diff
@@ -2513,5 +2513,12 @@ def from_pandas(obj):
             "Got %s" % type(obj)
         )
 
+def merge(left, right, *args, **kwargs):
+    return left.merge(right, *args, **kwargs)
+
+# a bit of fanciness to inject doctstring with left parameter
+merge_doc = DataFrame.merge.__doc__
```

Comment:
I'm not exactly happy with this bit of code but didn't want to go overboard with manipulating docstrings.  For reference, pandas has built some decorators to handle top-level and class level docstrings of the same function:https://github.com/pandas-dev/pandas/blob/master/pandas/core/reshape/merge.py#L38

Reply:
cc @randerzander since I know you were investigating this as well